### ISSUE TITLE
502 returned when issuing multiple "docker pull" in parallel against VIC ova

### ISSUE THREAD

**Issue by reasonerjt** [goharbor/harbor#3784]
The intermittent 502 errors in proxy.log:

```
Dec 11 01:10:28 172.18.0.1 proxy[393]: 10.133.221.64 - "HEAD /v2/default-project/cc100mb24/manifests/latest HTTP/1.1" 401 0 "-" "Go-http-client/1.1" 0.006 0.006 .
Dec 11 01:10:33 172.18.0.1 proxy[393]: 10.133.221.64 - "GET /v2/default-project/cc100mb24/manifests/latest HTTP/1.1" 200 3259 "-" "vic/v1.2.1-rc2" 0.814 0.814 .
Dec 11 01:10:34 172.18.0.1 proxy[393]: 10.133.221.64 - "GET /v2/default-project/cc100mb24/manifests/latest HTTP/1.1" 200 740 "-" "vic/v1.2.1-rc2" 1.253 1.253 .
Dec 11 01:50:53 172.18.0.1 proxy[393]: 10.133.221.64 - "HEAD /v2/default-project/cc100mb24/manifests/latest HTTP/1.1" 401 0 "-" "Go-http-client/1.1" 0.013 0.013 .
Dec 11 01:51:00 172.18.0.1 proxy[393]: 10.133.221.64 - "GET /v2/default-project/cc100mb24/manifests/latest HTTP/1.1" 502 0 "-" "vic/v1.2.1-rc2" 1.829 1.829 .
Dec 11 01:51:09 172.18.0.1 proxy[393]: 10.133.221.64 - "GET /v2/default-project/cc100mb24/manifests/latest HTTP/1.1" 200 3259 "-" "vic/v1.2.1-rc2" 0.258 0.258 .
Dec 11 01:51:10 172.18.0.1 proxy[393]: 10.133.221.64 - "GET /v2/default-project/cc100mb24/manifests/latest HTTP/1.1" 200 740 "-" "vic/v1.2.1-rc2" 0.042 0.042 .
Dec 11 02:02:58 172.18.0.1 proxy[393]: 10.133.221.64 - "HEAD /v2/default-project/cc100mb24/manifests/latest HTTP/1.1" 502 0 "-" "Go-http-client/1.1" 0.018 0.018 .
Dec 11 05:26:23 172.18.0.1 proxy[393]: 10.133.221.64 - "HEAD /v2/default-project/cc100mb24/manifests/latest HTTP/1.1" 401 0 "-" "Go-http-client/1.1" 0.021 0.021 .
Dec 11 05:26:29 172.18.0.1 proxy[393]: 10.133.221.64 - "GET /v2/default-project/cc100mb24/manifests/latest HTTP/1.1" 200 3259 "-" "vic/v1.2.1-rc2" 1.188 1.188 .
Dec 11 05:26:34 172.18.0.1 proxy[393]: 10.133.221.64 - "GET /v2/default-project/cc100mb24/manifests/latest HTTP/1.1" 200 740 "-" "vic/v1.2.1-rc2" 0.083 0.083 .
Dec 11 05:43:33 172.18.0.1 proxy[393]: 10.133.221.64 - "HEAD /v2/default-project/cc100mb24/manifests/latest HTTP/1.1" 401 0 "-" "Go-http-client/1.1" 0.075 0.075 .
Dec 11 05:43:39 172.18.0.1 proxy[393]: 10.133.221.64 - "GET /v2/default-project/cc100mb24/manifests/latest HTTP/1.1" 200 3259 "-" "vic/v1.2.1-rc2" 1.303 1.303 .
Dec 11 05:43:42 172.18.0.1 proxy[393]: 10.133.221.64 - "GET /v2/default-project/cc100mb24/manifests/latest HTTP/1.1" 200 740 "-" "vic/v1.2.1-rc2" 0.232 0.232 .
Dec 11 06:05:07 172.18.0.1 proxy[393]: 10.133.221.64 - "HEAD /v2/default-project/cc100mb24/manifests/latest HTTP/1.1" 401 0 "-" "Go-http-client/1.1" 0.007 0.007 .
Dec 11 06:05:13 172.18.0.1 proxy[393]: 10.133.221.64 - "GET /v2/default-project/cc100mb24/manifests/latest HTTP/1.1" 200 3259 "-" "vic/v1.2.1-rc2" 0.858 0.858 .
Dec 11 06:05:17 172.18.0.1 proxy[393]: 10.133.221.64 - "GET /v2/default-project/cc100mb24/manifests/latest HTTP/1.1" 200 740 "-" "vic/v1.2.1-rc2" 0.120 0.120 .
This is so far only observed in VIC deployment, it should be related to the chatty API against Admiral to check project policy.
```
Let's investigate the root cause, this may be mitigated by tuning some configuration in the nginx instance.

However, in the log I noticed the request is taking so long to return, we may consider to open another issue and improve the performance.

**Comment by wy65701436** [comment]
Tried scenario without catch any failure:
1, 10 times to run the pull script with interceptor enable. 
1, 10 times to run the pull script without interceptor enable. 

But, there is an limition of reproduce steps, the docker client and Harbor are located in the same Esxi.  

Will continuing to investigate this issue.

**Comment by moooofly** [comment]
In my environment, I got the 502 issue as well (in `proxy.log`)

```
Feb  3 08:00:01 ip-172-18-0-1.cn-north-1.compute.internal proxy[13962]: 172.1.50.19 - "GET /service/token?account=admin&scope=repository%3Aanalytics-rls%2Foverhear-sink%3Apull&service=harbor-registry HTTP/1.1" 200 978 "-" "docker/1.11.2 go/go1.5.4 git-commit/b9f10c9 kernel/4.4.41-k8s os/linux arch/amd64 UpstreamClient(Go-http-client/1.1)" 0.082 0.082 .
Feb  3 08:00:01 ip-172-18-0-1.cn-north-1.compute.internal proxy[13962]: 172.1.49.153 - "GET /service/token?account=admin&scope=repository%3Aanalytics-rls%2Foverhear-sink%3Apull&service=harbor-registry HTTP/1.1" 200 977 "-" "docker/1.11.2 go/go1.5.4 git-commit/b9f10c9 kernel/4.4.41-k8s os/linux arch/amd64 UpstreamClient(Go-http-client/1.1)" 0.074 0.074 .


Feb  3 08:00:02 ip-172-18-0-1.cn-north-1.compute.internal proxy[13962]: 172.1.33.129 - "GET /v2/analytics-rls/overhear-sink/manifests/v0.9.4 HTTP/1.1" 200 7029 "-" "docker/1.11.2 go/go1.5.4 git-commit/b9f10c9 kernel/4.4.41-k8s os/linux arch/amd64 UpstreamClient(Go-http-client/1.1)" 0.193 0.193 .


Feb  3 08:00:02 ip-172-18-0-1.cn-north-1.compute.internal proxy[13962]: 172.1.33.129 - "GET /v2/analytics-rls/overhear-sink/blobs/sha256:57f7c70b00e3b05838669dbc68e7fd9b8d5e499e20411adbec2bed2b4b97f751 HTTP/1.1" 307 501 "-" "docker/1.11.2 go/go1.5.4 git-commit/b9f10c9 kernel/4.4.41-k8s os/linux arch/amd64 UpstreamClient(Go-http-client/1.1)" 0.013 0.013 .
Feb  3 08:00:02 ip-172-18-0-1.cn-north-1.compute.internal proxy[13962]: 172.1.33.129 - "GET /v2/analytics-rls/overhear-sink/blobs/sha256:e411516a08c3bd0b3b66198b0de8fac780d202a5032a86e2615fdc217c8609cb HTTP/1.1" 307 501 "-" "docker/1.11.2 go/go1.5.4 git-commit/b9f10c9 kernel/4.4.41-k8s os/linux arch/amd64 UpstreamClient(Go-http-client/1.1)" 0.014 0.014 .
Feb  3 08:00:02 ip-172-18-0-1.cn-north-1.compute.internal proxy[13962]: 172.1.33.129 - "GET /v2/analytics-rls/overhear-sink/blobs/sha256:8a28109a801d003c5acd993f9c091a61db16d2de9d8a9784f48041cd319a13ea HTTP/1.1" 307 501 "-" "docker/1.11.2 go/go1.5.4 git-commit/b9f10c9 kernel/4.4.41-k8s os/linux arch/amd64 UpstreamClient(Go-http-client/1.1)" 0.014 0.014 .
Feb  3 08:00:02 ip-172-18-0-1.cn-north-1.compute.internal proxy[13962]: 172.1.33.129 - "GET /v2/analytics-rls/overhear-sink/blobs/sha256:bdb292d2f2cd3157ac4c4b88abb0c7fb65b9067313c0f8734ff539af537bc4c5 HTTP/1.1" 307 501 "-" "docker/1.11.2 go/go1.5.4 git-commit/b9f10c9 kernel/4.4.41-k8s os/linux arch/amd64 UpstreamClient(Go-http-client/1.1)" 0.025 0.025 .


Feb  3 08:00:02 ip-172-18-0-1.cn-north-1.compute.internal proxy[13962]: 172.1.50.119 - "GET /v2/ HTTP/1.1" 401 87 "-" "docker/1.11.2 go/go1.5.4 git-commit/b9f10c9 kernel/4.4.41-k8s os/linux arch/amd64 UpstreamClient(Go-http-client/1.1)" 0.003 0.003 .


Feb  3 08:00:02 ip-172-18-0-1.cn-north-1.compute.internal proxy[13962]: 172.1.33.129 - "GET /v2/analytics-rls/overhear-sink/blobs/sha256:fbfecac059709b43d03cec45df0d40cd564e8422a3caf240981a9c0df9a31c21 HTTP/1.1" 307 501 "-" "docker/1.11.2 go/go1.5.4 git-commit/b9f10c9 kernel/4.4.41-k8s os/linux arch/amd64 UpstreamClient(Go-http-client/1.1)" 0.004 0.004 .


Feb  3 08:00:02 ip-172-18-0-1.cn-north-1.compute.internal proxy[13962]: 172.1.50.119 - "GET /service/token?account=admin&scope=repository%3Aanalytics-rls%2Foverhear-sink%3Apull&service=harbor-registry HTTP/1.1" 200 978 "-" "docker/1.11.2 go/go1.5.4 git-commit/b9f10c9 kernel/4.4.41-k8s os/linux arch/amd64 UpstreamClient(Go-http-client/1.1)" 0.029 0.029 .


Feb  3 08:00:02 ip-172-18-0-1.cn-north-1.compute.internal proxy[13962]: 172.1.50.19 - "GET /v2/analytics-rls/overhear-sink/manifests/v0.9.4 HTTP/1.1" 200 7029 "-" "docker/1.11.2 go/go1.5.4 git-commit/b9f10c9 kernel/4.4.41-k8s os/linux arch/amd64 UpstreamClient(Go-http-client/1.1)" 0.250 0.250 .
Feb  3 08:00:02 ip-172-18-0-1.cn-north-1.compute.internal proxy[13962]: 172.1.49.153 - "GET /v2/analytics-rls/overhear-sink/manifests/v0.9.4 HTTP/1.1" 200 7029 "-" "docker/1.11.2 go/go1.5.4 git-commit/b9f10c9 kernel/4.4.41-k8s os/linux arch/amd64 UpstreamClient(Go-http-client/1.1)" 0.249 0.249 .


Feb  3 08:00:02 ip-172-18-0-1.cn-north-1.compute.internal proxy[13962]: 172.1.50.19 - "GET /v2/analytics-rls/overhear-sink/blobs/sha256:e12c678537aee9a1a1be8197da115e7c4d01f2652344f492a50ca8def9993d1e HTTP/1.1" 307 501 "-" "docker/1.11.2 go/go1.5.4 git-commit/b9f10c9 kernel/4.4.41-k8s os/linux arch/amd64 UpstreamClient(Go-http-client/1.1)" 0.016 0.016 .
Feb  3 08:00:02 ip-172-18-0-1.cn-north-1.compute.internal proxy[13962]: 172.1.50.19 - "GET /v2/analytics-rls/overhear-sink/blobs/sha256:bdb292d2f2cd3157ac4c4b88abb0c7fb65b9067313c0f8734ff539af537bc4c5 HTTP/1.1" 307 501 "-" "docker/1.11.2 go/go1.5.4 git-commit/b9f10c9 kernel/4.4.41-k8s os/linux arch/amd64 UpstreamClient(Go-http-client/1.1)" 0.011 0.011 .



Feb  3 08:00:02 ip-172-18-0-1.cn-north-1.compute.internal proxy[13962]: 172.1.49.153 - "GET /v2/analytics-rls/overhear-sink/blobs/sha256:4a97aa5ee463940ef5bc4cd08e1991f7d3a182d2e7f9bed866625b28a6c31866 HTTP/1.1" 502 0 "-" "docker/1.11.2 go/go1.5.4 git-commit/b9f10c9 kernel/4.4.41-k8s os/linux arch/amd64 UpstreamClient(Go-http-client/1.1)" 0.006 0.006 .
Feb  3 08:00:02 ip-172-18-0-1.cn-north-1.compute.internal proxy[13962]: 172.1.49.153 - "GET /v2/analytics-rls/overhear-sink/blobs/sha256:bdb292d2f2cd3157ac4c4b88abb0c7fb65b9067313c0f8734ff539af537bc4c5 HTTP/1.1" 502 0 "-" "docker/1.11.2 go/go1.5.4 git-commit/b9f10c9 kernel/4.4.41-k8s os/linux arch/amd64 UpstreamClient(Go-http-client/1.1)" 0.009 0.009 .
Feb  3 08:00:02 ip-172-18-0-1.cn-north-1.compute.internal proxy[13962]: 172.1.50.19 - "GET /v2/analytics-rls/overhear-sink/blobs/sha256:8d9ed335b7dbe095ecfbbfe0857d07971283db0119f7a4aa490f9cbe06187335 HTTP/1.1" 307 501 "-" "docker/1.11.2 go/go1.5.4 git-commit/b9f10c9 kernel/4.4.41-k8s os/linux arch/amd64 UpstreamClient(Go-http-client/1.1)" 0.014 0.014 .
Feb  3 08:00:02 ip-172-18-0-1.cn-north-1.compute.internal proxy[13962]: 172.1.49.153 - "GET /v2/analytics-rls/overhear-sink/blobs/sha256:8189e1876ae3cab5274e2da07c31445491d1c7bd1aae3cfb524573df6158af49 HTTP/1.1" 502 0 "-" "docker/1.11.2 go/go1.5.4 git-commit/b9f10c9 kernel/4.4.41-k8s os/linux arch/amd64 UpstreamClient(Go-http-client/1.1)" 0.006 0.006 .



Feb  3 08:00:02 ip-172-18-0-1.cn-north-1.compute.internal proxy[13962]: 172.1.50.19 - "GET /v2/analytics-rls/overhear-sink/blobs/sha256:5a6dcc67a16a729c0a52e5150d45ccb1e27950085ab8e7e63ac61b5502121979 HTTP/1.1" 307 501 "-" "docker/1.11.2 go/go1.5.4 git-commit/b9f10c9 kernel/4.4.41-k8s os/linux arch/amd64 UpstreamClient(Go-http-client/1.1)" 0.017 0.017 .
Feb  3 08:00:02 ip-172-18-0-1.cn-north-1.compute.internal proxy[13962]: 172.1.49.153 - "GET /v2/analytics-rls/overhear-sink/blobs/sha256:6e38a30b5306d3381034f19474b82d6635512af09b516a855a15dd87c7e857db HTTP/1.1" 307 501 "-" "docker/1.11.2 go/go1.5.4 git-commit/b9f10c9 kernel/4.4.41-k8s os/linux arch/amd64 UpstreamClient(Go-http-client/1.1)" 0.013 0.013 .
Feb  3 08:00:02 ip-172-18-0-1.cn-north-1.compute.internal proxy[13962]: 172.1.49.153 - "GET /v2/analytics-rls/overhear-sink/blobs/sha256:4a97aa5ee463940ef5bc4cd08e1991f7d3a182d2e7f9bed866625b28a6c31866 HTTP/1.1" 307 501 "-" "docker/1.11.2 go/go1.5.4 git-commit/b9f10c9 kernel/4.4.41-k8s os/linux arch/amd64 UpstreamClient(Go-http-client/1.1)" 0.004 0.004 .
Feb  3 08:00:02 ip-172-18-0-1.cn-north-1.compute.internal proxy[13962]: 172.1.49.153 - "GET /v2/analytics-rls/overhear-sink/blobs/sha256:8189e1876ae3cab5274e2da07c31445491d1c7bd1aae3cfb524573df6158af49 HTTP/1.1" 307 501 "-" "docker/1.11.2 go/go1.5.4 git-commit/b9f10c9 kernel/4.4.41-k8s os/linux arch/amd64 UpstreamClient(Go-http-client/1.1)" 0.005 0.005 .
Feb  3 08:00:02 ip-172-18-0-1.cn-north-1.compute.internal proxy[13962]: 172.1.50.119 - "GET /v2/analytics-rls/overhear-sink/manifests/v0.9.4 HTTP/1.1" 200 7029 "-" "docker/1.11.2 go/go1.5.4 git-commit/b9f10c9 kernel/4.4.41-k8s os/linux arch/amd64 UpstreamClient(Go-http-client/1.1)" 0.134 0.134 .
Feb  3 08:00:02 ip-172-18-0-1.cn-north-1.compute.internal proxy[13962]: 172.1.50.119 - "GET /v2/analytics-rls/overhear-sink/blobs/sha256:6fcd5d005ecfab86dd716406a83dbc2f817e26ff826e4f7db21701171a701718 HTTP/1.1" 307 501 "-" "docker/1.11.2 go/go1.5.4 git-commit/b9f10c9 kernel/4.4.41-k8s os/linux arch/amd64 UpstreamClient(Go-http-client/1.1)" 0.014 0.014 .
Feb  3 08:00:02 ip-172-18-0-1.cn-north-1.compute.internal proxy[13962]: 172.1.50.119 - "GET /v2/analytics-rls/overhear-sink/blobs/sha256:bdb292d2f2cd3157ac4c4b88abb0c7fb65b9067313c0f8734ff539af537bc4c5 HTTP/1.1" 307 501 "-" "docker/1.11.2 go/go1.5.4 git-commit/b9f10c9 kernel/4.4.41-k8s os/linux arch/amd64 UpstreamClient(Go-http-client/1.1)" 0.024 0.024 .
Feb  3 08:00:02 ip-172-18-0-1.cn-north-1.compute.internal proxy[13962]: 172.1.50.119 - "GET /v2/analytics-rls/overhear-sink/blobs/sha256:5a6dcc67a16a729c0a52e5150d45ccb1e27950085ab8e7e63ac61b5502121979 HTTP/1.1" 307 501 "-" "docker/1.11.2 go/go1.5.4 git-commit/b9f10c9 kernel/4.4.41-k8s os/linux arch/amd64 UpstreamClient(Go-http-client/1.1)" 0.021 0.021 .
Feb  3 08:00:02 ip-172-18-0-1.cn-north-1.compute.internal proxy[13962]: 172.1.50.119 - "GET /v2/analytics-rls/overhear-sink/blobs/sha256:0e13ccade694a2ab51939af312ea81dbb1e0658d8de407764789b8b3068c02fa HTTP/1.1" 307 501 "-" "docker/1.11.2 go/go1.5.4 git-commit/b9f10c9 kernel/4.4.41-k8s os/linux arch/amd64 UpstreamClient(Go-http-client/1.1)" 0.024 0.024 .
Feb  3 08:00:02 ip-172-18-0-1.cn-north-1.compute.internal proxy[13962]: 172.1.50.19 - "GET /v2/analytics-rls/overhear-sink/blobs/sha256:6fcd5d005ecfab86dd716406a83dbc2f817e26ff826e4f7db21701171a701718 HTTP/1.1" 307 501 "-" "docker/1.11.2 go/go1.5.4 git-commit/b9f10c9 kernel/4.4.41-k8s os/linux arch/amd64 UpstreamClient(Go-http-client/1.1)" 0.016 0.016 .
Feb  3 08:00:02 ip-172-18-0-1.cn-north-1.compute.internal proxy[13962]: 172.1.50.119 - "GET /v2/analytics-rls/overhear-sink/blobs/sha256:449966edb3748c4ae615355b58d5466e019d28367333db315144e67a91280070 HTTP/1.1" 307 501 "-" "docker/1.11.2 go/go1.5.4 git-commit/b9f10c9 kernel/4.4.41-k8s os/linux arch/amd64 UpstreamClient(Go-http-client/1.1)" 0.007 0.007 .
Feb  3 08:00:02 ip-172-18-0-1.cn-north-1.compute.internal proxy[13962]: 172.1.50.119 - "GET /v2/analytics-rls/overhear-sink/blobs/sha256:851ab8636be50592b4e27d72831bbfa0e834461bb98cd2cc7f31546fbeed8b62 HTTP/1.1" 307 501 "-" "docker/1.11.2 go/go1.5.4 git-commit/b9f10c9 kernel/4.4.41-k8s os/linux arch/amd64 UpstreamClient(Go-http-client/1.1)" 0.010 0.010 .
Feb  3 08:00:02 ip-172-18-0-1.cn-north-1.compute.internal proxy[13962]: 172.1.50.119 - "GET /v2/analytics-rls/overhear-sink/blobs/sha256:6e38a30b5306d3381034f19474b82d6635512af09b516a855a15dd87c7e857db HTTP/1.1" 307 501 "-" "docker/1.11.2 go/go1.5.4 git-commit/b9f10c9 kernel/4.4.41-k8s os/linux arch/amd64 UpstreamClient(Go-http-client/1.1)" 0.006 0.006 .
Feb  3 08:00:02 ip-172-18-0-1.cn-north-1.compute.internal proxy[13962]: 172.1.50.19 - "GET /v2/analytics-rls/overhear-sink/blobs/sha256:0e13ccade694a2ab51939af312ea81dbb1e0658d8de407764789b8b3068c02fa HTTP/1.1" 307 501 "-" "docker/1.11.2 go/go1.5.4 git-commit/b9f10c9 kernel/4.4.41-k8s os/linux arch/amd64 UpstreamClient(Go-http-client/1.1)" 0.009 0.009 .
Feb  3 08:00:02 ip-172-18-0-1.cn-north-1.compute.internal proxy[13962]: 172.1.50.19 - "GET /v2/analytics-rls/overhear-sink/blobs/sha256:851ab8636be50592b4e27d72831bbfa0e834461bb98cd2cc7f31546fbeed8b62 HTTP/1.1" 307 501 "-" "docker/1.11.2 go/go1.5.4 git-commit/b9f10c9 kernel/4.4.41-k8s os/linux arch/amd64 UpstreamClient(Go-http-client/1.1)" 0.004 0.004 .
Feb  3 08:00:02 ip-172-18-0-1.cn-north-1.compute.internal proxy[13962]: 172.1.50.119 - "GET /v2/analytics-rls/overhear-sink/blobs/sha256:8189e1876ae3cab5274e2da07c31445491d1c7bd1aae3cfb524573df6158af49 HTTP/1.1" 307 501 "-" "docker/1.11.2 go/go1.5.4 git-commit/b9f10c9 kernel/4.4.41-k8s os/linux arch/amd64 UpstreamClient(Go-http-client/1.1)" 0.004 0.004 .
Feb  3 08:00:02 ip-172-18-0-1.cn-north-1.compute.internal proxy[13962]: 172.1.50.19 - "GET /v2/analytics-rls/overhear-sink/blobs/sha256:449966edb3748c4ae615355b58d5466e019d28367333db315144e67a91280070 HTTP/1.1" 307 501 "-" "docker/1.11.2 go/go1.5.4 git-commit/b9f10c9 kernel/4.4.41-k8s os/linux arch/amd64 UpstreamClient(Go-http-client/1.1)" 0.016 0.016 .
Feb  3 08:00:02 ip-172-18-0-1.cn-north-1.compute.internal proxy[13962]: 172.1.50.119 - "GET /v2/analytics-rls/overhear-sink/blobs/sha256:4a97aa5ee463940ef5bc4cd08e1991f7d3a182d2e7f9bed866625b28a6c31866 HTTP/1.1" 307 501 "-" "docker/1.11.2 go/go1.5.4 git-commit/b9f10c9 kernel/4.4.41-k8s os/linux arch/amd64 UpstreamClient(Go-http-client/1.1)" 0.006 0.006 .
Feb  3 08:00:02 ip-172-18-0-1.cn-north-1.compute.internal proxy[13962]: 172.1.50.119 - "GET /v2/analytics-rls/overhear-sink/blobs/sha256:f43747a08c2f7b8c68823302fd6fe3d925838655187bf309531c9da398ecef6c HTTP/1.1" 307 501 "-" "docker/1.11.2 go/go1.5.4 git-commit/b9f10c9 kernel/4.4.41-k8s os/linux arch/amd64 UpstreamClient(Go-http-client/1.1)" 0.004 0.004 .
Feb  3 08:00:02 ip-172-18-0-1.cn-north-1.compute.internal proxy[13962]: 172.1.50.19 - "GET /v2/analytics-rls/overhear-sink/blobs/sha256:6e38a30b5306d3381034f19474b82d6635512af09b516a855a15dd87c7e857db HTTP/1.1" 307 501 "-" "docker/1.11.2 go/go1.5.4 git-commit/b9f10c9 kernel/4.4.41-k8s os/linux arch/amd64 UpstreamClient(Go-http-client/1.1)" 0.008 0.008 .
Feb  3 08:00:02 ip-172-18-0-1.cn-north-1.compute.internal proxy[13962]: 172.1.50.119 - "GET /v2/analytics-rls/overhear-sink/blobs/sha256:c8780f1e490e4af8f8c3ffb8b1a1fd322dfe39078e44a38460d2fe8e5e484f27 HTTP/1.1" 307 501 "-" "docker/1.11.2 go/go1.5.4 git-commit/b9f10c9 kernel/4.4.41-k8s os/linux arch/amd64 UpstreamClient(Go-http-client/1.1)" 0.004 0.004 .
Feb  3 08:00:02 ip-172-18-0-1.cn-north-1.compute.internal proxy[13962]: 172.1.50.119 - "GET /v2/analytics-rls/overhear-sink/blobs/sha256:2c273a375607a3bed5d9d5e8205c2b60a7ed3015e4f643939ab7b91d2f94746a HTTP/1.1" 307 501 "-" "docker/1.11.2 go/go1.5.4 git-commit/b9f10c9 kernel/4.4.41-k8s os/linux arch/amd64 UpstreamClient(Go-http-client/1.1)" 0.004 0.004 .
```

I find that 

- there is tree `502`s
- the return time of each `502` is just taking a little
- the hashes of three `502`s blobs appear as three `307`s blobs nearby.

so, I think this is maybe not a real problem, but it dose trigger my warning system.

**Comment by moooofly** [comment]
root case: After sending HTTP request, if the response from Harbor costs more the 60s (nginx timeout value by default), will lead to "502" problem.

update `nginx.conf` settings as follow can solve this issue:

```
  server {
    listen 443 ssl;
    ...
    proxy_connect_timeout 600;      -- add
    proxy_send_timeout    600;      -- add
    proxy_read_timeout    600;      -- add
    send_timeout          600;      -- add
    ...
}
```

**Comment by moooofly** [comment]
UPDATE: I still encounter 502 error now and then with above nginx config adjustment (I'm sure it solves problems of some situation).

At present, I doubt that the 502 error may be related to nginx worker process num. By default, `worker_processes auto;` and I have only two logical cpu core.

**Comment by moooofly** [comment]
UPDATE: 2018-07-11

After scaling up the machine running harbor (from 2 to 8 logical CPU cores), the 502 error does not show up for now. 

I will keep watching it for a while.

**Comment by reasonerjt** [comment]
Thanks @moooofly for the update!
